### ISSUE TITLE
Zeek 5.0.8 for Fedora 38

### ISSUE THREAD

**Issue by LawrenceRRogers** [zeek/zeek#2974]
Zeek 5.0.8 does not build for Fedora 38 for x86_64

Part of the problem is solved with this patch:
===============================================================================
```
diff --color -c -r -N zeek-5.0.8-original/auxil/broker/caf/libcaf_core/caf/detail/ripemd_160.hpp zeek-5.0.8/auxil/broker/caf/libcaf_core/caf/detail/ripemd_160.hpp
*** zeek-5.0.8-original/auxil/broker/caf/libcaf_core/caf/detail/ripemd_160.hpp	2023-04-11 15:00:53.000000000 -0400
--- zeek-5.0.8/auxil/broker/caf/libcaf_core/caf/detail/ripemd_160.hpp	2023-04-21 09:46:18.560662429 -0400
***************
*** 42,47 ****
--- 42,48 ----
  
  #include <array>
  #include <string>
+ #include <cstdint>
  
  #include "caf/detail/core_export.hpp"
  
diff --color -c -r -N zeek-5.0.8-original/auxil/broker/caf/libcaf_core/caf/telemetry/metric_type.hpp zeek-5.0.8/auxil/broker/caf/libcaf_core/caf/telemetry/metric_type.hpp
*** zeek-5.0.8-original/auxil/broker/caf/libcaf_core/caf/telemetry/metric_type.hpp	2023-04-11 15:00:53.000000000 -0400
--- zeek-5.0.8/auxil/broker/caf/libcaf_core/caf/telemetry/metric_type.hpp	2023-04-21 18:41:35.719460165 -0400
***************
*** 4,9 ****
--- 4,11 ----
  
  #pragma once
  
+ #include <cstdint>
+ 
  namespace caf::telemetry {
  
  enum class metric_type : uint8_t {
```
===============================================================================

In addition, this error remains:

===============================================================================
```
[ 28%] Built target reproc
[ 28%] Building CXX object auxil/spicy/spicy/3rdparty/reproc/reproc++/CMakeFiles/reproc++.dir/src/reproc.cpp.o
In file included from /home/repoman/rpmbuild/BUILD/zeek-5.0.8/auxil/spicy/spicy/3rdparty/reproc/reproc++/src/reproc.cpp:1:
/home/repoman/rpmbuild/BUILD/zeek-5.0.8/auxil/spicy/spicy/3rdparty/reproc/reproc++/include/reproc++/reproc.hpp:95:5: error: declaration of 'reproc::options::<unnamed struct> reproc::options::env' changes meaning of 'env' [-Wchanges-meaning]
   95 |   } env = {};
      |     ^~~
/home/repoman/rpmbuild/BUILD/zeek-5.0.8/auxil/spicy/spicy/3rdparty/reproc/reproc++/include/reproc++/reproc.hpp:91:5: note: used here to mean 'class reproc::env'
   91 |     env::type behavior;
      |     ^~~
In file included from /home/repoman/rpmbuild/BUILD/zeek-5.0.8/auxil/spicy/spicy/3rdparty/reproc/reproc++/include/reproc++/reproc.hpp:11:
/home/repoman/rpmbuild/BUILD/zeek-5.0.8/auxil/spicy/spicy/3rdparty/reproc/reproc++/include/reproc++/env.hpp:8:7: note: declared here
    8 | class env : public detail::array {
      |       ^~~
/home/repoman/rpmbuild/BUILD/zeek-5.0.8/auxil/spicy/spicy/3rdparty/reproc/reproc++/include/reproc++/reproc.hpp:107:5: error: declaration of 'reproc::options::<unnamed struct> reproc::options::redirect' changes meaning of 'redirect' [-Wchanges-meaning]
  107 |   } redirect = {};
      |     ^~~~~~~~
/home/repoman/rpmbuild/BUILD/zeek-5.0.8/auxil/spicy/spicy/3rdparty/reproc/reproc++/include/reproc++/reproc.hpp:102:5: note: used here to mean 'struct reproc::redirect'
  102 |     redirect err;
      |     ^~~~~~~~
/home/repoman/rpmbuild/BUILD/zeek-5.0.8/auxil/spicy/spicy/3rdparty/reproc/reproc++/include/reproc++/reproc.hpp:68:8: note: declared here
   68 | struct redirect {
      |        ^~~~~~~~
make[3]: *** [auxil/spicy/spicy/3rdparty/reproc/reproc++/CMakeFiles/reproc++.dir/build.make:76: auxil/spicy/spicy/3rdparty/reproc/reproc++/CMakeFiles/reproc++.dir/src/reproc.cpp.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:5009: auxil/spicy/spicy/3rdparty/reproc/reproc++/CMakeFiles/reproc++.dir/all] Error 2
make[1]: *** [Makefile:166: all] Error 2
make[1]: Leaving directory '/home/repoman/rpmbuild/BUILD/zeek-5.0.8/build'
make: *** [Makefile:15: all] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.IBPVJE (%build)
===============================================================================
```
Thanks for your help in addressing these issues.

Larry Rogers
lrr@cert.org

**Comment by ckreibich** [comment]
Thank you for flagging! You beat us to our monthly CI upgrade, which is going to pull in Fedora 38.

**Comment by LawrenceRRogers** [comment]
Thanks!

From: Tim Wojtulewicz ***@***.***>
Date: Monday, April 24, 2023 at 9:59 AM
To: zeek/zeek ***@***.***>
Cc: Larry Rogers ***@***.***>, Author ***@***.***>
Subject: Re: [zeek/zeek] Zeek 5.0.8 for Fedora 38 (Issue #2974)

This error is fixed upstream in Spicy already. We just need to pull the fix back into the release branch for zeek 5.0.

**Comment by LawrenceRRogers** [comment]
Thanks!

From: Christian Kreibich ***@***.***>
Date: Monday, April 24, 2023 at 9:43 AM
To: zeek/zeek ***@***.***>
Cc: Larry Rogers ***@***.***>, Author ***@***.***>
Subject: Re: [zeek/zeek] Zeek 5.0.8 for Fedora 38 (Issue #2974)

Thank you for flagging! You beat us to our monthly CI upgrade, which is going to pull in Fedora 38.

**Comment by awelzel** [comment]
> This error is fixed upstream in Spicy already. We just need to pull the fix back into the release branch for zeek 5.0.

@LawrenceRRogers - if I understand things correctly, you can invoke `CXXFLAGS=-Wno-changes-meaning ./configure ...` on Fedora as a workaround for now until a permanent fix for the 5.0.x maintenance release.

**Comment by timwoj** [comment]
I merged the above PR into the release branch and bumped our opensuse-tumbleweed CI host to pick up gcc 13 there. It looks like the spicy issue are resolved. I'll see about a broker patch to fix the issues you originally reported as well.

**Comment by timwoj** [comment]
https://github.com/zeek/zeek/pull/3009 should have resolved all of the rest of the GCC 13 issues on the release/5.0 branch, plus added a CI build to cover it. I'm going to close this issue. The fixes will be in the next 5.0 release.

**Comment by LawrenceRRogers** [comment]
OK – thanks. Hopefully what I provided helped you get ‘er going sooner rather than later.

Much appreciated.
Larry Rogers

Principal Engineer
 Security Operations Team
CERT Division 
 Software Engineering Institute 
Carnegie Mellon University 
4500 Fifth Avenue 
Pittsburgh, PA 15213, USA 

Please note that I am semi-retired and am working part-time.

PGP Key ID: 0x16441F81 (RSA)
Fingerprint: C851 9DA0 0BC8 2B6F E0C4 DB15 CF3B 92D4
http://pgp.mit.edu:11371/pks/lookup?op=get&search=0x16441F81

PGP Key ID: 0xF8546B01 (DSA)
Fingerprint: 002C B634 E913 9EF1 EEE3 8B4F 8313 51B0 F854 6B01
http://pgp.mit.edu:11371/pks/lookup?op=get&search=0xF8546B01

URL: http://www.sei.cmu.edu/about/people/profile.cfm?id=rogers_13248

***@***.***

NOTICE: This message and any files transmitted with it may contain information that is privileged or confidential. It is intended only for the use of the individual or entity to whom they are addressed. If you are not the intended recipient, you are hereby notified that any use, dissemination, distribution or copying of this communication and its attachments is strictly prohibited. If you have received this message in error, please notify the sender immediately and delete all copies of this message and all attachments. If you made a hard copy, please return it via the U.S. Postal Service. Thank you.



From: Tim Wojtulewicz ***@***.***>
Date: Thursday, May 4, 2023 at 4:05 PM
To: zeek/zeek ***@***.***>
Cc: Larry Rogers ***@***.***>, Mention ***@***.***>
Subject: Re: [zeek/zeek] Zeek 5.0.8 for Fedora 38 (Issue #2974)

#3009<https://github.com/zeek/zeek/pull/3009> should have resolved all of the rest of the GCC 13 issues on the release/5.0 branch, plus added a CI build to cover it. I'm going to close this issue. The fixes will be in the next 5.0 release.
